lib/board.rb

```diff
@@ -0,0 +1,43 @@
+class Board
+  attr_accessor :cells
+  def initialize
+    self.reset!
+  end
+
+  def reset!
+    self.cells=Array.new(9, " ")
+  end
+
+  def display
+    board=self.cells
+      puts " #{board[0]} | #{board[1]} | #{board[2]} "
+      puts "-----------"
+      puts " #{board[3]} | #{board[4]} | #{board[5]} "
+      puts "-----------"
+      puts " #{board[6]} | #{board[7]} | #{board[8]} "
+    end
+
+  def position(n)
+    self.cells[n.to_i-1]
+  end
+
+  def full?
+    !self.cells.include?(" ")
+  end
+
+  def turn_count
+    9 - self.cells.count(" ")
+  end
+
+  def taken?(n)
+    position(n) != " "
+  end
+
+  def valid_move?(n)
+    n.to_i <=9 && n.to_i >=1 && !taken?(n)
```

Comment:
You can simplify checking the valid integer range, and make it more semantic, by using [`Comparable#between?`](http://ruby-doc.org/core-2.2.3/Comparable.html#method-i-between-3F).
`n.between?(1,9)`